# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/gtk/nsWaylandDisplay.cpp
# Commit: b4fee14fe36a
# Full Hash: b4fee14fe36ae084fb392de2a40bcbd9a226bec3
# Author: Martin Stransky <stransky@redhat.com>
# Date: 2020-06-14 09:24:22
# Regressor Bug: 1629788
# File Overlap Count: 1
# Description:
#   Bug 1629788 [Wayland] Add widget.wayland-dmabuf-video-textures.enabled preference, r=jhorak
#   
#   - widget.wayland-dmabuf-video-textures.enabled controls dmabuf texture backend for decoded video frames.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D78293
# ==============================================================================

diff -r 8a463700d169 -r b4fee14fe36a widget/gtk/nsWaylandDisplay.cpp
--- a/widget/gtk/nsWaylandDisplay.cpp	Sat Jun 13 18:38:38 2020 +0000
+++ b/widget/gtk/nsWaylandDisplay.cpp	Sat Jun 13 18:38:45 2020 +0000
@@ -498,6 +498,7 @@
       !StaticPrefs::widget_wayland_dmabuf_basic_compositor_enabled() &&
       !StaticPrefs::widget_wayland_dmabuf_textures_enabled() &&
 #endif
+      !StaticPrefs::widget_wayland_dmabuf_video_textures_enabled() &&
       !StaticPrefs::widget_wayland_dmabuf_webgl_enabled() &&
       !StaticPrefs::widget_wayland_dmabuf_vaapi_enabled()) {
     // Disabled by user, just quit.
@@ -535,6 +536,10 @@
 bool nsWaylandDisplay::IsDMABufBasicEnabled() { return false; }
 bool nsWaylandDisplay::IsDMABufTexturesEnabled() { return false; }
 #endif
+bool nsWaylandDisplay::IsDMABufVideoTexturesEnabled() {
+  return IsDMABufEnabled() &&
+         StaticPrefs::widget_wayland_dmabuf_video_textures_enabled();
+}
 bool nsWaylandDisplay::IsDMABufWebGLEnabled() {
   return IsDMABufEnabled() &&
          StaticPrefs::widget_wayland_dmabuf_webgl_enabled();